spring-integration-stomp/src/main/java/org/springframework/integration/stomp/inbound/StompInboundChannelAdapter.java

```diff
@@ -0,0 +1,273 @@
+
+	private final Lock destinationLock = new ReentrantLock();
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private volatile StompSession stompSession;
+
+	private volatile Class<?> payloadType = String.class;
+
+	private volatile HeaderMapper<StompHeaders> headerMapper = new StompHeaderMapper();
+
+	private volatile MessageChannel errorChannel;
+
+	public StompInboundChannelAdapter(StompSessionManager stompSessionManager, String... destinations) {
+		Assert.notNull(stompSessionManager, "'stompSessionManager' is required.");
+		if (destinations != null) {
+			for (String destination : destinations) {
+				Assert.hasText(destination, "'destinations' must not have empty strings.");
+				this.destinations.add(destination);
+			}
+		}
+		this.stompSessionManager = stompSessionManager;
+	}
+
+	public void setPayloadType(Class<?> payloadType) {
+		Assert.notNull(payloadType, "'payloadType' must not be null.");
+		this.payloadType = payloadType;
+	}
+
+	@Override
+	public void setErrorChannel(MessageChannel errorChannel) {
+		super.setErrorChannel(errorChannel);
+		this.errorChannel = errorChannel;
+	}
+
+	public void setHeaderMapper(HeaderMapper<StompHeaders> headerMapper) {
+		Assert.notNull(headerMapper, "'headerMapper' must not be null.");
+		this.headerMapper = headerMapper;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	@ManagedAttribute
+	public String[] getDestinations() {
+		this.destinationLock.lock();
+		try {
+			return this.destinations.toArray(new String[this.destinations.size()]);
+		}
+		finally {
+			this.destinationLock.unlock();
+		}
+	}
+
+	/**
+	 * Add a destination (or destinations) to the subscribed list and subscribe it.
+	 * @param destination The destinations.
+	 */
+	@ManagedOperation
+	public void addDestination(String... destination) {
+		Assert.notNull(destination, "'destination' cannot be null");
+		this.destinationLock.lock();
+		try {
+			for (String d : destination) {
+				if (this.destinations.add(d)) {
+					if (this.logger.isDebugEnabled()) {
+						logger.debug("Removed '" + d + "' from subscriptions.");
+					}
+					subscribeDestination(d);
+				}
+			}
+		}
+		finally {
+			this.destinationLock.unlock();
+		}
+	}
+
+	/**
+	 * Remove a destination (or destinations) from the subscribed list and unsubscribe it.
+	 * @param destination The destinations.
+	 */
+	@ManagedOperation
+	public void removeDestination(String... destination) {
+		this.destinationLock.lock();
+		try {
+			for (String d : destination) {
+				if (this.destinations.remove(d)) {
+					if (this.logger.isDebugEnabled()) {
+						logger.debug("Removed '" + d + "' from subscriptions.");
+					}
+					StompSession.Subscription subscription = this.subscriptions.get(d);
+					if (subscription != null) {
+						subscription.unsubscribe();
+					}
+				}
```

Comment:
Not sure if it makes sense.
Just copied/pasted from the `MqttPahoMessageDrivenChannelAdapter`.
Or I just don't understand what you mean here...